packages/cli/src/Server.ts

```diff
@@ -2200,3 +2201,10 @@ async function getExecutionsCount(countFilter: IDataObject): Promise<{ count: nu
 	const count = await Db.collections.Execution!.count(countFilter);
 	return { count, estimate: false };
 }
+
+async function generateInstanceId(){
+	const encryptionKey = await UserSettings.getEncryptionKey();
+	const hash = encryptionKey ? createHash('sha256').update(encryptionKey).digest('hex') : undefined;
```

Comment:
to be on the safe side, can you take the first 16 characters of `encryptionKey` before hashing it?

Reply:
Applied the suggestion. Theoretically, we may find 2 or more encryption keys with the same last 16 characters which may cause a conflict.